tools/api-linter/src/error.rs

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use anyhow::{Context, Result};
+use owo_colors::{OwoColorize, Stream};
+use pest::Position;
+use rustdoc_types::Span;
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::fmt;
+use std::path::{Path, PathBuf};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum RefType {
```

Comment:
surprised that `RefType` lives alongside errors but context gets its own module

Reply:
I hope the doc comments I added clarify this a bit.